source/extensions/http/header_validators/envoy_default/config_overrides.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "envoy/server/factory_context.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Http {
+namespace HeaderValidators {
+namespace EnvoyDefault {
+
+class ConfigOverrides {
+public:
+  ConfigOverrides() = default;
+  ConfigOverrides(Server::Configuration::ServerFactoryContext& server_context)
```

Comment:
nit: Consider passing in a `const Envoy::Runtime::Snapshot&` instead of the full server context. I *think* this might allow the various members to be const and in tests, we can construct `ConfigOverrides` objects from MockSnapshots with the right defaults set.

Reply:
Done